### PR TITLE
Static analyzer issue fixes

### PR DESCRIPTION
**Pull request by spbnick**
This is a bunch of fixes for some new issues found by Coverity and Clang in v3.0.13, compared to v3.0.4.

Please feel free to pick and choose as you see fit.

### CODE REVIEW

File: src/main/evaluate.c [hunk 1]
```diff
@@ -167,6 +167,10 @@ static int cond_do_regex(REQUEST *request, fr_cond_t const *c,
 	rad_assert(lhs_type == PW_TYPE_STRING);
 	rad_assert(lhs != NULL);
 
+	if (lhs_type != PW_TYPE_STRING || lhs == NULL) {
+		return -1;
+	}
+
```
Comment (alandekok) [inline src/main/evaluate.c]: The assertions a few lines up should also check for this.  Either Coverity is missing the assertions, OR we should have only one check.  i.e. there's not much point in having assertions *and* run-time checks.

**Comment by alandekok** [comment]
I'll push some simpler changes, and then manually pull over some of the patches.  I would rather not have 3-4 patches that all touch the same code, and I'd rather not have both asserts and run-time checks.

**Comment by spbnick** [comment]
Thanks, Alan. Yeah, we run Coverity and clang-scan on non-debug build, because that's what we package, so assertions are ignored by Coverity.

**Comment by spbnick** [comment]
Alan, are you done with these? Asking because I'd like to make sure I picked and backported all the fixes you took from here.

**Comment by alandekok** [comment]
I've pushed a bunch of fixes over.  I'll be doing the rest in a few days

**Comment by nwkrad** [comment]
Can one of the admins verify this patch?

**Comment by nwkrad** [comment]
Can one of the admins verify this patch?

**Comment by arr2036** [comment]
@alandekok did you merge all the fixes, can this be closed?

**Comment by alandekok** [comment]
There are some fixes remaining, but I don't recall which ones.  I'll find time to look at it again.

**Comment by alandekok** [comment]
These have all been merged.

**Comment by spbnick** [comment]
Thanks, Alan!

**Pull request closed without merging**